rust/relay/src/main.rs

```diff
@@ -542,6 +553,23 @@ fn fmt_human_throughput(mut throughput: f64) -> String {
     format!("{throughput:.2} TB/s")
 }
 
+/// Factory fn for [`is_healthy`].
+fn make_is_healthy(
+    last_heartbeat_sent: Arc<Mutex<Option<Instant>>>,
+) -> impl Fn() -> bool + Clone + Send + Sync + 'static {
+    move || is_healthy(last_heartbeat_sent.clone())
+}
+
+fn is_healthy(last_heartbeat_sent: Arc<Mutex<Option<Instant>>>) -> bool {
```

Comment:
We should add a comment like:

"This will always return `true` during normal operation, a `false` result is only possible in case of a busy-loop that prevents shutdown in case of a partition"

Reply:
> "This will always return `true` during normal operation, a `false` result is only possible in case of a busy-loop that prevents shutdown in case of a partition"

But that is not true! The `is_healthy` function has no idea why we aren't sending heartbeats. The only correct comment here would be: This will return `false` if we haven't sent a heartbeat in longer than `MAX_PARTITION_TIME`.

There are at least two reasons why we aren't sending heartbeats:

- We are actually partitioned
- We have a bug in the event-loop that makes us not process anything regarding the portal

Reply:
> We have a bug in the event-loop that makes us not process anything regarding the portal

I am a bit torn on whether we should add a comment for this _here_ because it might get stale because `is_healthy` has nothing to do with the event-loop per-se.